### PR TITLE
Feature #8 employee

### LINKED ISSUE #8: Create Employe Model/Controller
# REQUIREMENTS
Create a class and all the methods for an employe

# Acceptance Criteria
List all required acceptance criteria
- [ ] The employe can check in and check out with his private number

### PR DESCRIPTION
**Pull request by AlejandroVI99**
# Description
What did you implemented/Why did you implemented this:
 - Explain the changes you’ve made. 
 - We created the model of the employee, also, we created the migration where we add the attributes of the employee.
 - We created the views edit, index and new for employee.
 - We created the controller for the employee where we add the methods edit, create new, update, destroy and index.
 - Also we created a partial for the form.
 
- We fixed a rubycritic problem 
 
  
  ## Screenshots 
image
image

- for the button destroy we added message to confirm if we really want to delete the employee.
image

### CODE REVIEW

File: kata06/app/controllers/employees_controller.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+class EmployeesController < ApplicationController
+    def index
+        @employees = Employee.all
+    end
```
Comment (juliocabrera820) [inline kata06/app/controllers/employees_controller.rb:4]: you can add a blank line below

File: kata06/app/controllers/employees_controller.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    def new
+        @employee = Employee.new
+    end
+
+    def create
+        data = params.require(:employee).permit(:name, :email, :posicion, :num_empleado, :private_num)
```
Comment (juliocabrera820) [inline kata06/app/controllers/employees_controller.rb:10]: You can create a private method called employee_params like this:

```ruby
def employee_params
  params.require(:employee).permit(:name, :email, :posicion. :num_empleado, :private_num)
end
```

File: kata06/app/controllers/employees_controller.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        Employee.create(data)
```
Comment (juliocabrera820) [inline kata06/app/controllers/employees_controller.rb:11]: you can also use the save method and compare the returned value to render a successful or error message

File: kata06/app/controllers/employees_controller.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        redirect_to employees_path
+    end
+
+    def edit
+        @employee = Employee.find(params[:id])
+    end
+
+    def update
+        @employee = Employee.find(params[:id])
+        data = params.require(:employee).permit(:name, :email, :posicion, :num_empleado, :private_num)
```
Comment (juliocabrera820) [inline kata06/app/controllers/employees_controller.rb:22]: You can create a private method called employee_params like this:

```ruby
def employee_params
  params.require(:employee).permit(:name, :email, :posicion. :num_empleado, :private_num)
end
```

File: kata06/app/controllers/employees_controller.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+    def create
+        data = params.require(:employee).permit(:name, :email, :posicion, :num_empleado, :private_num)
+        Employee.create(data)
+
+        redirect_to employees_path
+    end
+
+    def edit
+        @employee = Employee.find(params[:id])
```
Comment (juliocabrera820) [inline kata06/app/controllers/employees_controller.rb:17]: You can create a private method called employee_params like this:

```ruby
def employee_params
  params.require(:employee).permit(:name, :email, :posicion. :num_empleado, :private_num)
end
```

File: kata06/app/controllers/employees_controller.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    end
+
+    def update
+        @employee = Employee.find(params[:id])
```
Comment (juliocabrera820) [inline kata06/app/controllers/employees_controller.rb:21]: You can create a private method called set_employee like this:

```ruby
def set_employee
  @employee = Employee.find(params[:id])
end
```

File: kata06/app/controllers/employees_controller.rb [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+        data = params.require(:employee).permit(:name, :email, :posicion, :num_empleado, :private_num)
+        @employee.update(data)
+
+        redirect_to employees_path
+    end
+
+    def destroy
+        @employee = Employee.find(params[:id])
```
Comment (juliocabrera820) [inline kata06/app/controllers/employees_controller.rb:29]: You can create a private method called set_employee like this:

```ruby
def set_employee
  @employee = Employee.find(params[:id])
end
```

File: kata06/app/views/employees/index.html.erb [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+      <th>Email</th>
+      <th>Posicion</th>
+      <th>Num empleado</th>
+      <th>Private num</th>
+      <th colspan="3"></th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <% @employees.each do |e| %>
```
Comment (juliocabrera820) [inline kata06/app/views/employees/index.html.erb:14]: you can be more descriptive with the name e

File: kata06/app/controllers/employees_controller.rb [hunk 9]
```diff
@@ -0,0 +1,35 @@
+class EmployeesController < ApplicationController
```
Comment (juliocabrera820) [inline kata06/app/controllers/employees_controller.rb:1]: below this line, you can use the hook before_action like this:

```ruby
before_action :set_employe, only: i%[show edit update destroy]
```

**Review by juliocabrera820 — DISMISSED** [review]
LGTM 👍🏼

**Review by SampabloC — DISMISSED** [review]
LGTM

File: kata06/app/controllers/employees_controller.rb (hunk 1 above)
Comment (AlejandroVI99) [reply kata06/app/controllers/employees_controller.rb:4]: where?

File: kata06/app/controllers/employees_controller.rb (hunk 2 above)
Comment (AlejandroVI99) [reply kata06/app/controllers/employees_controller.rb:10]: ok! understood c:

File: kata06/app/controllers/employees_controller.rb (hunk 3 above)
Comment (AlejandroVI99) [reply kata06/app/controllers/employees_controller.rb:11]: ok! understood c:

File: kata06/app/controllers/employees_controller.rb (hunk 4 above)
Comment (AlejandroVI99) [reply kata06/app/controllers/employees_controller.rb:22]: i will edit that for my next PR!

File: kata06/app/controllers/employees_controller.rb (hunk 5 above)
Comment (AlejandroVI99) [reply kata06/app/controllers/employees_controller.rb:17]: i will edit that for my next PR!

File: kata06/app/controllers/employees_controller.rb (hunk 6 above)
Comment (AlejandroVI99) [reply kata06/app/controllers/employees_controller.rb:21]: as the employee_params method, i wil edit that in my next PR!

File: kata06/app/controllers/employees_controller.rb (hunk 7 above)
Comment (AlejandroVI99) [reply kata06/app/controllers/employees_controller.rb:29]: as the employee_params method, i wil edit that in my next PR!

File: kata06/app/views/employees/index.html.erb (hunk 8 above)
Comment (AlejandroVI99) [reply kata06/app/views/employees/index.html.erb:14]: ok! i will put employee!

File: kata06/app/controllers/employees_controller.rb (hunk 9 above)
Comment (AlejandroVI99) [reply kata06/app/controllers/employees_controller.rb:1]: if you can explain me what that line means i will appreciated! (the "before_action" i already understand what that means)

**Review by OmarRoBra — DISMISSED** [review]

**Review by xFerchoVJ — APPROVED** [review]
LGTM!

**Review by martinmt-mx — APPROVED** [review]
All good :)

**Review by SEVERO96 — APPROVED** [review]

**Pull request merged**